app/controllers/alchemy/admin/pages_controller.rb

```diff
@@ -81,7 +81,10 @@ def create
       def edit
         # fetching page via before filter
         if page_is_locked?
-          flash[:notice] = Alchemy.t('This page is locked', name: @page.locker_name)
+          flash[:warning] = Alchemy.t('This page is locked', name: @page.locker_name)
+          redirect_to admin_pages_path
+        elsif !@page.editable_by?(current_alchemy_user)
```

Comment:
This branch should never be reached because `CanCan` makes sure that `authorize!` is called before every action and will redirect the user to the Dashboard if she's not authorized. 

Your tests are correct, but this branch is never hit, and we have not been able to create a situation where it is. We propose deleting this `elsif` branch and keeping the specs that are there. Sorry for the confusion! :)